packages/api/src/task/scheduler.ts

```diff
@@ -400,6 +401,30 @@ export class TaskScheduler {
         },
       });
     }
+    if (task.status.phase === "completed" && task.outputAssetId) {
```

Comment:
We should avoid adding more business logic to this `updateTask` helper, as it should handle mostly the "storage update" and any consequential things like webhooks. We probably did a poor job in keeping that true here over time, but in this specific case we're handling the completion phase of a task, which has a better place in the code today.

So the right place for this logic would be inside `processTaskEvent` instead, specifically under the [`"upload"` task case](https://github.com/livepeer/livepeer-com/blob/ad8d909a045b536ffa7cc15955e828e6e4e6aa5d/packages/api/src/task/scheduler.ts#L124-L126) which is what we use for recordings afaik. There it would also have more guarantee that it is processing an actual completion event for the task, and not any update that could trigger accidental/duplicate webhooks here.

Would you mind moving this to `processTaskEvent`?

Reply:
Updated